### PR TITLE
Add Login for OIDC client

### PR DESCRIPTION
**Pull request by sbrunner**
See JIRA issue: [GSGGR-152](https://jira.camptocamp.com/browse/GSGGR-152).
[Examples](https://camptocamp.github.io/ngeo/refs/pull/9408/merge/examples/)
[Storybook](https://camptocamp.github.io/ngeo/refs/pull/9408/merge/storybook/)
[API help](https://camptocamp.github.io/ngeo/refs/pull/9408/merge/api/apihelp/apihelp.html)
[API documentation](https://camptocamp.github.io/ngeo/refs/pull/9408/merge/apidoc/)

[GSGGR-152]: https://camptocamp.atlassian.net/browse/GSGGR-152?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/auth/FormElement.ts [hunk 1]
```diff
@@ -79,6 +83,16 @@ export default class GmfAuthForm extends GmfBaseElement {
...
+      setInterval(() => {
+        this.currentApplicationUrl = window.location.href;
+        const applicationUrl = this.loginInfoMessage
+          ? this.currentApplicationUrl
+          : this.initialApplicationUrl;
+        const params = new URLSearchParams({
+          came_from: applicationUrl,
+        });
+        this.openIdConnectUrl = `${this.openIdConnectBaseUrl}?${params.toString()}`;
+      }, 200),
```
Comment (sbrunner) [inline src/auth/FormElement.ts:95]: TODO: remove this hack by creating a store for the permalink

**Comment by sbrunner** [comment]
Linked to: https://github.com/camptocamp/c2cgeoportal/pull/11316

File: src/auth/FormElement.ts [hunk 2]
```diff
@@ -168,7 +189,7 @@ export default class GmfAuthForm extends GmfBaseElement {
                       </div>
                       <div class="form-group">
                         <input
-                          ?hidden="${!this.allowPasswordChange}"
+                          ?hidden="${!(this.allowPasswordChange && this.gmfUser.login_type != 'oidc')}"
```
Comment (ger-benjamin) [inline src/auth/FormElement.ts:192]:
```suggestion
                          ?hidden="${!(this.allowPasswordChange && this.gmfUser.login_type !== 'oidc')}"
```

File: src/auth/FormElement.ts [hunk 3]
```diff
@@ -196,58 +217,62 @@ export default class GmfAuthForm extends GmfBaseElement {
...
-                      <div class="login-spinner">
-                        <i class="fa fa-spin"
-                          >${
-                            // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-                            unsafeSVG(loadingSvg)
-                          }</i
-                        >
-                      </div>
-                    `
+        ? this.gmfUser.login_type == 'oidc'
```
Comment (ger-benjamin) [inline src/auth/FormElement.ts:220]:
```suggestion
        ? this.gmfUser.login_type === 'oidc'
```

**Review by ger-benjamin — APPROVED** [review]
Looks good

**Pull request merged**